### PR TITLE
Vertical fd operator support on hybrid domains

### LINKED ISSUE #142: Add functionality allowing for broadcasting vertical operators over 1D, 2D extruded domains

### PR DESCRIPTION
**Pull request by jakebolewski**
Closes #142

### CODE REVIEW

**Comment by jakebolewski** [comment]
bors try

**Comment by bischtob** [comment]
Can we r+ this @simonbyrne ?

File: test/hybrid.jl [hunk 1]
```diff
@@ -54,13 +91,13 @@ end
         # horizontal divergence operator applied to all levels
         hdiv = Operators.Divergence()
         @. dudt = hdiv(u * Geometry.Cartesian1Vector(1.0))
```
Comment (bischtob) [inline test/hybrid.jl:93]: Do we support this horizontal operator with Cartesian13Vectors like in the test above? It seems like we should be able to apply a spectral divergence to the full field, @simonbyrne, no?

File: test/hybrid.jl [hunk 2]
```diff
@@ -17,31 +17,68 @@ import ClimaCore.Domains.Geometry: Cartesian2DPoint
...
+
+    hv_center_space, hv_face_space = hvspace_2D()
+    V =
+        Geometry.Cartesian13Vector.(
+            zeros(Float64, hv_face_space),
+            ones(Float64, hv_face_space),
+        )
 
+    function rhs!(dudt, u, _, t)
+        A = Operators.AdvectionC2C(
```
Comment (bischtob) [inline test/hybrid.jl:57]: Here it seems that that `AdvectionC2C` operates on `Cartesian13Vector`, should we expect the same behavior from the spectral operators, @simonbyrne ?

Comment (bischtob) [reply test/hybrid.jl:57]: Ok, I can see how we can leverage the implicit Ekman example for the vertical velocity equation here.

File: test/hybrid.jl (hunk 1 above)
Comment (bischtob) [reply test/hybrid.jl:93]: Or at least we'd want to advect a `Cartesian3Vector` by a `Cartesian1Vector`?

**Pull request merged**